### DISCUSSION TITLE
Draw a graph containing a tree structure for easy viewing.

### DISCUSSION THREAD

**Discussion by kancve** [ndsev/zserio#419] [Ideas]
I refer to the xml extension to implement the json extension, generate a json file, and use [echarts ](https://echarts.apache.org/examples/en/editor.html?c=tree-polyline)to render the tree structure diagram, the effect is quite good. If allowed, I will push the source code to this repository.

![echarts](https://user-images.githubusercontent.com/29945891/190634895-feb491b9-6c4c-46c1-962e-a24256d4d541.gif)

Best wishes!

**Comment by fklebert** [comment]
Love the idea... looks really cool. 👍 
Maybe you can add the code to your fork? Then we can check a corresponding PR and see whether we want to maintain such a feature in the future.
Again: Nice work :)

**Reply by kancve** [reply]
Thank you for your praise. 
The code has been pushed to my [branch](https://github.com/kancve/zserio/tree/maven_build). It is just a simple example at the moment, I will gradually improve it later.
Instead of writing an ant build file with json extension, I use the [build.bash](https://github.com/kancve/zserio/blob/maven_build/mvn_build/build.bash) script to build the project, it just does some work of copying the source code and executes the mvn command.

**Comment by kancve** [comment]
I deployed the final rendering result [here](https://kancve.github.io/zserio-echarts/), and the data to be displayed is [here](https://github.com/kancve/zserio-echarts/tree/master/public/data).
